### DISCUSSION TITLE
Add headers?

### DISCUSSION THREAD

**Discussion by halloleo** [saulpw/visidata#2368] [Q&A]
I have plenty of CSVs without a header line, e.g.:

```
2023-03-12,16,28.5
2023-03-13,15.1,23
2023-03-14,15.9,26.2
2023-03-15,16.2,29.1
```

For the moment I generate headers _before_ I look at the file in vd:

     $ csvcut -H data.csv | vd

and then I have in `vd`:

| a          | b    | c    |
| ---------- | ---- | ---- |
| 2023-03-12 | 16   | 28.5 |
| 2023-03-13 | 15.1 | 23   |
| 2023-03-14 | 15.9 | 26.2 |
| 2023-03-15 | 16.2 | 29.1 |

but is there an easy way/command line option to tell `vd`  to generate some default headers?

**Comment by ajkerrigan** [comment]
I don't know of a way to generate a header row like that. You can use `--header 0` to tell vd that there isn't a header in your data, but that'll leave the column names blank. And there are commands to bulk-rename columns (like `rename-cols-selected` and `rename-cols-row`), but those use row data to feed into headers.

The closest thing I can think of is doing it with a custom command in your `~/.visidatarc` file

```python
import string
from visidata import BaseSheet


@BaseSheet.api
def addDefaultHeader(sheet):
    sheet.options.header = 0
    sheet.reload()
    sheet.vd.sync()
    sheet.updateColNames(string.ascii_lowercase.split(), sheet.visibleCols)


BaseSheet.addCommand(
    "",
    "add-default-header",
    "sheet.addDefaultHeader()",
    "add a default header row",
)
```

Which would let you run a `add-default-header` command (optionally with a shortcut key) inside vd to generate that default header line. ...but your csvcut way seems easier :-/. Someone else may have a better idea.

**Comment by saulpw** [comment]
You could do `Cgsgiq`.

**Reply by ajkerrigan** [reply]
Haha that's way less work :sweat_smile:

**Reply by saulpw** [reply]
It doesn't make useful Python identifiers though, which is probably what they wanted :)

**Reply by halloleo** [reply]
Just back to this issue...

`Cgsgiq` is a very cool way to do it. Thanks.

However for me it "eats" the first data line of the CSV file.
